workflow/metrics/metrics.go

```diff
@@ -153,6 +155,23 @@ func (m *Metrics) StopRealtimeMetricsForKey(key string) {
 		return
 	}
 
+	realtimeMetrics := m.workflows[key]
+	for _, metric := range realtimeMetrics {
+		if realtimeMetric, ok := m.customMetrics[metric]; ok {
+			realtimeMetric.completed = true
+			m.customMetrics[metric] = realtimeMetric
```

Comment:
What is this line doing, shouldn't it already be in this array?

Reply:
The value of `customMetrics` map is `struct` type, not a pointer type, so `realtimeMetric` is actually a copy and I need to update it to the map again.

Reply:
Grr, I hate that pattern. Understood.